File: Data Structures & Algorithms/reverse-bits/submission-0.py
class Solution:
    def reverseBits(self, n: int) -> int:
        result = 0
        i = 0
        while n > 0:
            if n & 1 == 1:
                result = result | (1 << (31 - i))
            
            n = n >> 1
            i += 1
        
        return result

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
   public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> result(nums.size(), 1);
        int product = 1;
        for (int n : nums) {
            product *= n;
        }

        int leftProduct = 1;
        int rightProduct = 1;
        for (int i = 0; i < nums.size(); i++) {
            result[i] *= leftProduct;
            leftProduct *= nums[i];
        }

        for (int i = nums.size() - 1; i >= 0; i--) {
            result[i] *= rightProduct;
            rightProduct *= nums[i];
        }

        return result;
    }
};
